app/Http/Middleware/ProductMiddleware.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Auth;
+
+class ProductMiddleware
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request Request
+     * @param \Closure                 $next    Next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (auth()->check() && auth()->user()->role==2) {
```

Comment:
[rank B] did you define for user role as constant? Please use it instead hard code as number.